python_modules/dagster/dagster/_core/definitions/observe.py

```diff
@@ -12,22 +15,27 @@
     from ..execution.execute_in_process_result import ExecuteInProcessResult
 
 
+@deprecated_param(
+    param="source_assets", breaking_version="2.0.0", additional_warn_text="Use `assets` instead."
+)
 def observe(
-    source_assets: Sequence[SourceAsset],
+    assets: Optional[Sequence[Union[AssetsDefinition, SourceAsset]]] = None,
```

Comment:
why make this optional and default to `None`?

Reply:
After Sandy pointed out this isn't public I changed this to just `Sequence[Union[AssetsDefinition, SourceAsset]]`.